src/wpsmon.c

```diff
@@ -429,6 +434,11 @@ void parse_wps_settings(const u_char *packet, struct pcap_pkthdr *header, char *
 
 					char* vendor = get_vendor_string(get_ap_vendor(bssid));
 					char* sane_ssid = sanitize_string(ssid);
+					if(show_utf8_ssid)
+					    {
+					        if(verifyssid(ssid))
+					            strcpy(sane_ssid,ssid);
```

Comment:
this is potentially dangerous since sane_ssid is malloced. right now it will work (except when malloc fails) since sanitize_string will always return a string that's of equal length or larger, but this detail might be overlooked in the future...